alibi/explainers/integrated_gradients.py

```diff
@@ -428,6 +429,7 @@ def explain(self,
             else:
                 grads_b = _gradients_input(self.model,
                                            tf.dtypes.cast(paths_b, self.input_dtype), target_b)
+                # TODO: is casting the right thing to do? Will loose information if input types differ
```

Comment:
Think that's fine, input types for a given model / layer should be consistent anyway.